frontend/src/components/Graph.tsx

```diff
@@ -154,6 +154,9 @@ export default class Graph extends React.Component<GraphProps, GraphState> {
           // deviating from the explicit layout provided by dagre.
           if (i === 1) {
             const sourceNode = graph.node(edgeInfo.v);
+            if (sourceNode === undefined) {
```

Comment:
I believe this (and below) can be removed with the fix to `WorkflowParser`

Reply:
done